DropwizardHelloWorld/src/test/java/com/example/helloworld/service/RetrieveMessageServiceTest.java

```diff
@@ -0,0 +1,42 @@
+package com.example.helloworld.service;
+
+import lombok.extern.java.Log;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Created by ccardona on 4/12/17.
+ */
+public class RetrieveMessageServiceTest {
+
+    final static Logger log = LoggerFactory.getLogger(RetrieveMessageServiceTest.class);
+
+    private RetrieveMessageService messageService;
+    private static final String MESSAGE_ONE = "Testing one";
+    private static final String MESSAGE_TWO = "Testing two";
+
+    @Before
+    public void init() {
+        MessageGeneratorInterface mockitoOne = Mockito.mock(MessageOne.class);
+        Mockito.when(mockitoOne.generateMessage()).thenReturn(MESSAGE_ONE);
+        MessageGeneratorInterface mockitoTwo = Mockito.mock(MessageTwo.class);
+        Mockito.when(mockitoTwo.generateMessage()).thenReturn(MESSAGE_TWO);
+        messageService = new RetrieveMessageService(mockitoOne, mockitoTwo);
+    }
+
+    @Test
+    public void testGetMyCoolMessage() {
+        log.info("Starting test testGetMyCoolMessage");
+        Assert.assertTrue(messageService.getMyCoolMessage(1).equals(MESSAGE_ONE));
```

Comment:
This code is bug prone. What if `messageService.getMyCoolMessage(1)` returns `null`? You can eliminate the risk of getting a `NullPointerException` by rewriting the expression as: `MESSAGE_ONE.equals(messageService.getMyCoolMessage(1))`

A better option is to use `Assert.assertEquals()` for asserting equality. An even better option is to use `Assert.assertThat()` + [Hamcrest Matchers](https://dzone.com/articles/using-hamcrest-and-junit). `assertThat` increases readability of the test and generate better error messages when an assertion fails, which is more than enough reasons to use it, IMHO. If you are not yet convinced, you may want to take a look at the following blog post: https://objectpartners.com/2013/09/18/the-benefits-of-using-assertthat-over-other-assert-methods-in-unit-tests/